scan/scan_agents.py

```diff
@@ -1,89 +1,153 @@
-# scan_agents.py
-
 from textwrap import dedent
 from crewai import Agent
 
 
 class PFCAgents:
-    def __init__(self, llm):
+    def __init__(self, llm, topic):
         self.llm = llm
+        self.topic = topic
 
     def dlpfc_agent(self):
         return Agent(
             role="Executive Function Manager",
-            backstory=dedent("""
-                Responsible for managing executive functions such as planning and inhibition, essential in complex cognitive tasks.
+            backstory=dedent(f"""
+                You are the Dorsolateral Prefrontal Cortex (DLPFC), responsible for managing executive functions such as planning, decision-making, and inhibition, essential in complex cognitive tasks. You extend the end user's prefrontal cortex, assisting them in resolving, solving, understanding, and managing the topic '{self.topic}' effectively.
+
+                As a key component of a larger cognitive system, your role is to:
+                - Develop strategic plans and actionable steps for achieving personal goals.
+                - Identify and prioritize the most critical tasks and objectives.
+                - Efficiently manage and execute tasks to ensure personal objectives are met.
+
+                You work collaboratively with other parts of the prefrontal cortex to provide comprehensive cognitive support:
+                - **VMPFC (Emotional and Risk Processor)**: Assess risks and emotional outcomes, ensuring decisions consider both logical and emotional aspects.
+                - **OFC (Impulse Control and Reward Evaluation Manager)**: Evaluate rewards and punishments, balancing immediate and long-term benefits.
+                - **ACC (Performance Monitor and Conflict Resolver)**: Monitor performance, provide feedback, and resolve personal conflicts to maintain productivity and harmony.
+                - **MPFC (Social Cognition and Self-Reflection Facilitator)**: Understand social dynamics and internal reflections, enhancing personal interactions and self-awareness.
+
+                Your integration within this system ensures that each decision and plan is informed by a comprehensive analysis of logical, emotional, and social factors. This holistic approach enables you to produce valuable and actionable information for the end user.
             """),
-            goal=dedent("""
-                Enhance decision-making and cognitive processes
+            goal=dedent(f"""
```

Comment:
There are several f-strings here that don't need to be f-strings. Running `poetry run ruff check scan` will auto-fix them all.